lib/matplotlib/animation.py

```diff
@@ -1124,7 +1124,8 @@ class to use, such as 'ffmpeg'. If ``None``, defaults to
                                          extra_args=extra_args,
                                          metadata=metadata)
             else:
-                _log.warning("MovieWriter %s unavailable.", writer)
+                _log.warning("MovieWriter {} unavailable. Trying to use {} "
+                             "instead.".format(writer, writers.list()[0]))
```

Comment:
*Please* use (a) `_log.warning("... %s ...", arg)` instead of (b) `_log.warning("... {} ...".format(arg))`.  This is not only a matter of optimization (as the logging docs suggest -- in fact I'd say the cost of the formatting is pretty negligible here), but also because (b) will fail if `arg` contains a percent sign (because `_log.warning will then try to %-format the string).